bin/install_windows_10_sdk.ps1

```diff
@@ -0,0 +1,59 @@
+# Stop script execution when a non-terminating error occurs
+$ErrorActionPreference = "Stop"
+
+Write-Host "--- :windows: Installing Windows 10 SDK and Visual Studio Build Tools"
+
+$windowsSDKVersionFile = ".windows-10-sdk-version"
+if (-not (Test-Path $windowsSDKVersionFile)) {
+  Write-Output "[!] No Windows 10 SDK version file found at $windowsSDKVersionFile."
+  exit 1
+}
+
+$windows10SDKVersion = Get-Content $windowsSDKVersionFile
```

Comment:
Question: how does this `Get-Content` handles possible newlines?

Especially, if the version file ends with a newline at end of file, is that trimmed?
Or maybe it's not, but that doesn't cause any issues because when you then reference `$windows10SDKVersion` in other commands like `--add Microsoft.VisualStudio.Component.Windows10SDK.$windows10SDKVersion` that's when the value is trimmed as usage/substitution site?
What if the file contains 2 newlines at end of file? 🤔

Reply:
Great question! We clearly need to add something to strip them:

![image](https://github.com/user-attachments/assets/bfd24af9-1d29-473a-b975-8c79bc723e04)

https://buildkite.com/automattic/ci-toolkit-buildkite-plugin/builds/1007#01951c5c-320d-4346-8d45-3e9bda56ad1e

Reply:
Spent more time than I care to admit on this but got a bunch of tests for the version parsing

![image](https://github.com/user-attachments/assets/50d0046c-6eae-478d-81a4-7607e2ca0d96)

I'll clean up tomorrow https://github.com/Automattic/a8c-ci-toolkit-buildkite-plugin/pull/153